### PR TITLE
[SPARK-38732][SQL][TESTS] Test the error class: INCOMPARABLE_PIVOT_COLUMN

### PR DESCRIPTION
**Pull request by lvshaokang**
### What changes were proposed in this pull request?

I add a test case for the error class INCOMPARABLE_PIVOT_COLUMN in the QueryExecutionErrorsSuite.

### Why are the changes needed?

To improve test coverage, and record expected error messages in tests.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

By existing tests.

### CODE REVIEW

**Comment by lvshaokang** [comment]
@MaxGekk Thanks for your review.  I changed done.

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala [hunk 1]
```diff
@@ -398,4 +398,32 @@ class QueryExecutionErrorsSuite
         "\\(QueryExecutionErrorsSuite\\$\\$Lambda\\$\\d+/\\w+: \\(string, int\\) => string\\)",
       matchMsg = true)
   }
+
+  test("INCOMPARABLE_PIVOT_COLUMN: Pivot columns can't be comparable") {
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala:402]: Maybe, the columns can't be incomparable?
```suggestion
  test("INCOMPARABLE_PIVOT_COLUMN: an incomparable column of the map type") {
```

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala [hunk 2]
```diff
@@ -398,4 +398,32 @@ class QueryExecutionErrorsSuite
...
+    withSQLConf() {
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala:403]: Please, remove this.

**Review by MaxGekk — CHANGES_REQUESTED** [review]

**Review by MaxGekk — APPROVED** [review]
+1, LGTM. Merging to master.
Thank you, @lvshaokang.

**Pull request closed without merging**